isolated-jobs-test/src/main/java/com/hazelcast/jet/tests/isolated/IsolatedJobsStreamTest.java

```diff
@@ -0,0 +1,170 @@
+                DEFAULT_JOB_WINDOW_TUMBLE_IN_MINUTES);
+        sleepBetweenStreamRecordsMs = propertyInt("sleepBetweenStreamRecordsMs",
+                DEFAULT_SLEEP_BETWEEN_STREAM_RECORDS_MS);
+    }
+
+    @Override
+    protected void test(HazelcastInstance client, String name) throws Throwable {
+        long begin = System.currentTimeMillis();
+        String logPrefix = "[" + name + "] ";
+
+        UUID excludedMember = client.getCluster().getMembers().iterator().next().getUuid();
+        Job streamJob = JobDefinition.createStreamJob(jobWindowTumbleInMinutes, sleepBetweenStreamRecordsMs,
+                snapshotIntervalMs, client, excludedMember, name);
+        waitForJobStatus(streamJob, RUNNING);
+
+        long validationCount = 0;
+        while (System.currentTimeMillis() - begin < durationInMillis) {
+            if (getJobStatusWithRetry(streamJob) == JobStatus.FAILED) {
+                throw new RuntimeException(logPrefix + "Job failed: " + streamJob.getName());
+            }
+
+            sleepMinutes(sleepBeforeValidationInMinutes);
+
+            assertStreamSink(client, excludedMember, name);
+            clearSink(client, name);
+
+            validationCount++;
+            if (validationCount % logVerificationCountThreshold == 0) {
+                logger.info(logPrefix + "Validations count: " + validationCount);
+            }
+        }
+        logger.info(logPrefix + "Final validations count: " + validationCount);
+    }
+
+    @Override
+    protected void teardown(Throwable t) throws Exception {
+
+    }
+
+    private void clearSink(HazelcastInstance client, String jobName) {
+        client.getList(jobName).clear();
+    }
+
+    private void assertStreamSink(HazelcastInstance client, UUID excludedMember, String jobName) {
+        IList<List<UUID>> list = client.getList(jobName);
+        assertFalse("There was no records in the List from job " + jobName, list.isEmpty());
+        list.forEach(uuids -> {
+            boolean result = uuids.stream()
+                    .noneMatch(uuid -> uuid.equals(excludedMember));
+            if (!result) {
+                throw new AssertionError("Job (" + jobName + ") was executed on excluded member ("
+                        + excludedMember + ")." +
+                        "List of members from execution: " + uuids.stream()
+                        .map(UUID::toString)
+                        .collect(joining(","))
+                );
+            }
+        });
+    }
+
+    public static class JobDefinition {
+        public static Job createStreamJob(int jobWindowTumbleInMinutes, int sleepBetweenStreamRecordsMs,
+                                          int snapshotIntervalMs, HazelcastInstance client,
+                                          UUID excludedMember, String jobName) {
+            Pipeline p = Pipeline.create();
+            p.readFrom(Sources.createStreamSource(sleepBetweenStreamRecordsMs))
```

Comment:
Why do we use tumbling window here? It makes test much more complicated and it doesn't seem to be relevant to isolated job feature at all?

Reply:
Ok, simplified.